### PR TITLE
Code Review Request - Rework Edit Lesson Page

### PR DESCRIPTION
**Pull request by AshSensei**

### CODE REVIEW

File: client/src/components/conclusions-list.tsx
Comment (dan100110) [inline client/src/components/conclusions-list.tsx:64]: why the change in text?

File: client/src/components/expectations-list.tsx
Comment (dan100110) [inline client/src/components/expectations-list.tsx:284]: why remove the default text for the expectation

File: client/src/pages/lessons/edit.tsx
Comment (dan100110) [inline client/src/pages/lessons/edit.tsx:249]: A lot of CSS that I don't have enough expertise to review.  Best to get a review from either Aaron, or Dilan.  Maybe both for the initial one.

File: client/src/pages/lessons/edit.tsx
Comment (dan100110) [inline client/src/pages/lessons/edit.tsx:637]: what's the spec for this?  the expectation results accuracy shouldn't affect whether the training failed or succeeded.

File: client/src/pages/lessons/edit.tsx
Comment (dan100110) [inline client/src/pages/lessons/edit.tsx:880]: something about this smells, It feels like you're trying to do too much in the view portion of the codebase.  If this really is all in UI code, then consider breaking it down into smaller components

File: client/src/pages/lessons/edit.tsx
Comment (dan100110) [inline client/src/pages/lessons/edit.tsx:889]: remember to add data-cy tags for cypress testing

File: client/src/pages/lessons/edit.tsx
Comment (dan100110) [inline client/src/pages/lessons/edit.tsx:1]: this page is way to big and needs to be broken up into smaller components each with it's own handlers.  If possible some of the data manipulation should be moved to actions as well.

File: client/src/pages/lessons/edit.tsx
Comment (dan100110) [inline client/src/pages/lessons/edit.tsx:301]: again I'm not sure about the best practices regarding CSS,  It seems to me that the CSS should be defined separately from the functionality.

File: cypress/cypress/e2e/edit-lesson.cy.ts
Comment (dan100110) [inline cypress/cypress/e2e/edit-lesson.cy.ts:135]: Don't commit commented out lines,  It's better to rely on version control to recover deleted code.  as for the content.  If the fields no longer exist then remove it.  If they do still exist, but the values have changed then the tests should reflect that

File: cypress/cypress/e2e/edit-lesson.cy.ts
Comment (dan100110) [inline cypress/cypress/e2e/edit-lesson.cy.ts:154]: should have some way to test for images and video, but that may not be easy with the framework.

**Review by dan100110 — CHANGES_REQUESTED** [review]
lot of stuff to fix here even after fixing the cypress tests.  The big problem with the failing tests is that you've removed elements that they're expecting.  You'll have to address that.

File: client/src/components/conclusions-list.tsx
Comment (AshSensei) [reply client/src/components/conclusions-list.tsx:64]: We made an overhaul to a lot of the phrasing and to make it more aligned with what teacher's are used to seeing in their learning management systems. I understand how the example text could still be useful so if we wanted to push for keeping it in we can.

File: client/src/components/expectations-list.tsx
Comment (AshSensei) [reply client/src/components/expectations-list.tsx:284]: The default text for the expectation caused the user to have to manually highlight and delete/change the text whenever they created a new lesson. This could be a minor obstacle for teachers as usually when you let's say draft an email, there isn't example text already filled out and so by removing the default values we get rid of that minor hiccup for the user and make it more straightforward.

File: client/src/pages/lessons/edit.tsx
Comment (AshSensei) [reply client/src/pages/lessons/edit.tsx:1]: Got it. I can work on that.

**Review by AshSensei — COMMENTED** [review]
I'll work on breaking the page down more to be less bloaty since it has grown to be a bit too large. I will also make sure to note the comments of not committing comments and the tests.

**Review by dan100110 — COMMENTED** [review]
I'm okay with this, but Aaron's more knowledgeable about react, so I'll wait for him to approve it.

File: client/src/components/side-bar.tsx
Comment (aaronshiel) [inline client/src/components/side-bar.tsx:190]: remove log

File: client/src/components/side-bar.tsx
Comment (aaronshiel) [inline client/src/components/side-bar.tsx:219]: move this to a helpers file

File: client/src/components/side-bar.tsx
Comment (aaronshiel) [inline client/src/components/side-bar.tsx:156]: If possible, I'd move these CSSObjects and styled components to a separate file to clean up this component. This component is getting very bloated.

File: client/src/components/side-bar.tsx
Comment (aaronshiel) [inline client/src/components/side-bar.tsx:294]: move this into a helper function in a separate helper file, this code should reduce down to something like:

`const trainAiButtonColor = getTrainButtonColor(trainStatus)`

File: client/src/components/side-bar.tsx
Comment (aaronshiel) [inline client/src/components/side-bar.tsx:333]: also move this into a helper function (in separate helper function file), this if block should reduce down to something like:

`const lastTrainedString = getLastTrainedAtString(lessonUnderEdit)`

File: client/src/components/side-bar.tsx
Comment (aaronshiel) [inline client/src/components/side-bar.tsx:319]: For cleanliness, could move this to a separate constants file, especially useful if a list of the months is ever needed elsewhere.

File: client/src/pages/lessons/edit.tsx
Comment (aaronshiel) [inline client/src/pages/lessons/edit.tsx:821]: looks like this whole div for displaying "advanced features" could be moved into its own component (for better readability/maintainability/etc.)

File: client/src/pages/lessons/edit.tsx
Comment (aaronshiel) [inline client/src/pages/lessons/edit.tsx:686]: should move this anonymous function into its own separate named function. rule of thumb is if the anonymous function is more than 2-3 lines of code, move it into it's own name function. (apply this to other anonymous functions in the file)

File: client/src/pages/lessons/edit.tsx
Comment (aaronshiel) [inline client/src/pages/lessons/edit.tsx:894]: remove commented code

**Review by aaronshiel — CHANGES_REQUESTED** [review]

File: client/src/pages/lessons/edit.tsx
Comment (aaronshiel) [inline client/src/pages/lessons/edit.tsx:503]: The AdvancedFeatures component should be moved into it's own file. Overall I think the components in this file could be modularized a bit more

File: client/src/pages/lessons/edit.tsx
Comment (aaronshiel) [inline client/src/pages/lessons/edit.tsx:657]: For example, maybe move this "lesson format selector" into its own component file

**Review by aaronshiel — CHANGES_REQUESTED** [review]

File: client/src/components/advanced-features.tsx
Comment (aaronshiel) [inline client/src/components/advanced-features.tsx:1]: very nice and modularized! good work!

**Review by aaronshiel — APPROVED** [review]

**Review by dan100110 — APPROVED** [review]

**Pull request merged**
